### PR TITLE
Scrub names from messages in logging/telemetry

### PR DESCRIPTION
**Pull request by manodnyab**
## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.telemetry
+
+import org.junit.jupiter.api.Assertions.assertEquals
```
Comment (rli) [inline plugins/core/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtilsTest.kt:6]: wrong assertion type

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt [hunk 2]
```diff
@@ -200,3 +200,66 @@ val ALLOWED_CODE_EXTENSIONS = setOf(
     "yml",
     "zig"
 )
+
+fun scrubNames(messageToBeScrubbed: String, username: String? = null): String {
+    var scrubbedMessage = ""
+    val fileExtRegex = Regex("""\.[^./]+$""")
+    val slashdot = Regex("""^[~.]*[/\\]*""")
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt:207]: regex instances should be preserved and not created on every invocation

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt [hunk 3]
```diff
@@ -200,3 +200,66 @@ val ALLOWED_CODE_EXTENSIONS = setOf(
...
+
+    /** Allowlisted filepath segments. */
+    val keep = setOf(
+        "~", ".", "..", ".aws", "aws", "sso", "cache", "credentials", "config",
+        "Users", "users", "home", "tmp", "aws-toolkit-jetbrains"
+    )
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt:213]: move to top level so we don't need to recreate the set every time

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt [hunk 4]
```diff
@@ -200,3 +200,66 @@ val ALLOWED_CODE_EXTENSIONS = setOf(
     "yml",
     "zig"
 )
+
+fun scrubNames(messageToBeScrubbed: String, username: String? = null): String {
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryUtils.kt:204]: dont we need to pass in a user

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt [hunk 5]
```diff
@@ -182,7 +183,7 @@ class SsoAccessTokenProvider(
                 source = "accessToken",
                 result = Result.Failed,
                 reason = "Failed to write AccessToken to cache",
-                reasonDesc = e.message ?: e::class.java.name,
+                reasonDesc = e.message?.let { scrubNames(it) } ?: e::class.java.name,
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:186]: you could probably just make it handle the null case

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt (hunk 5 above)
Comment (manodnyab) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:187]: Will address it in a follow-up pr

**Pull request merged**